src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -2305,26 +2287,27 @@ void InitThermalAndFluxHistories(EnergyPlusData &state)
 
     // Perform other initializations that depend on the surface characteristics
     for (int CTFTermNum = 1; CTFTermNum <= state.dataHeatBal->MaxCTFTerms + 1; ++CTFTermNum) {
-        for (SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
-            if (!Surface(SurfNum).HeatTransSurf) continue; // Skip non-heat transfer surfaces
+        for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+            auto &surface = state.dataSurface->Surface(SurfNum);
+            if (!surface.HeatTransSurf) continue; // Skip non-heat transfer surfaces
             // Reset outside boundary conditions if necessary
-            if ((Surface(SurfNum).ExtBoundCond == ExternalEnvironment) || (Surface(SurfNum).ExtBoundCond == OtherSideCondModeledExt)) {
+            if ((surface.ExtBoundCond == DataSurfaces::ExternalEnvironment) || (surface.ExtBoundCond == DataSurfaces::OtherSideCondModeledExt)) {
                 state.dataHeatBalSurf->SurfOutsideTempHist(CTFTermNum)(SurfNum) = state.dataSurface->SurfOutDryBulbTemp(SurfNum);
-            } else if (Surface(SurfNum).ExtBoundCond == Ground) {
+            } else if (surface.ExtBoundCond == DataSurfaces::Ground) {
                 state.dataHeatBalSurf->SurfOutsideTempHist(CTFTermNum)(SurfNum) = state.dataEnvrn->GroundTemp;
-            } else if (Surface(SurfNum).ExtBoundCond == GroundFCfactorMethod) {
+            } else if (surface.ExtBoundCond == DataSurfaces::GroundFCfactorMethod) {
                 state.dataHeatBalSurf->SurfOutsideTempHist(CTFTermNum)(SurfNum) = state.dataEnvrn->GroundTempFC;
             }
             // Initialize the flux histories
             state.dataHeatBalSurf->SurfOutsideFluxHist(CTFTermNum)(SurfNum) =
-                state.dataConstruction->Construct(Surface(SurfNum).Construction).UValue *
+                state.dataConstruction->Construct(surface.Construction).UValue *
                 (state.dataHeatBalSurf->SurfOutsideTempHist(1)(SurfNum) - state.dataHeatBalSurf->SurfInsideTempHist(1)(SurfNum));
             state.dataHeatBalSurf->SurfInsideFluxHist(CTFTermNum)(SurfNum) = state.dataHeatBalSurf->SurfOutsideFluxHist(2)(SurfNum);
         }
     }
-    for (SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
-
-        if (!Surface(SurfNum).HeatTransSurf) continue; // Skip non-heat transfer surfaces
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
```

Comment:
There is a list of heat transfer surfaces. Do we want to create a list of ExtCavity surfaces also?